README.md

```diff
@@ -1,7 +1,124 @@
+
+### Models and Workflow of Vscode-Granite
+
+vscode-granite uses the Granite Code model, which is optimized for enterprise software development workflows and performs well across a range of coding tasks (e.g., code generation, fixing, and explanation), making it a versatile "all-around" code model.
+
+Granite Code comes in a wide range of sizes to fit your workstation's available resources. Generally, the bigger the model, the better the results.
+
+**Recommendation:** Model Size 20B for chat, 8B for tab code completion.
+
+For more details, refer to [Granite Code Models](https://github.com/ibm-granite/granite-code-models)
+
+ ### Workflow of Vscode-Granite
+
+ <p align="center">
+<img src="/media/workflow.png" alt="workflow" width="400"/>
+ </p>
+ 
+### Installation Prerequisites:
+
+  * OS: MacOS 
+  * DISK SPACE :Minimum 30 GB 
+  * latest [Visual Studio Code](https://code.visualstudio.com/)
+  * [Node.js](https://nodejs.org/) v20.0.0 or higher
+  * optional: [gh](https://cli.github.com/) to manage pull requests.
+  * Install [continu.dev](https://marketplace.visualstudio.com/items?itemName=Continue.continue) extention on Visual studio code
+
+### Steps run the extension
+  * clone this repository
+
+  <body>
+    <h4>How to install vs-granite on macOS/Linux</h4>
+    <p>Follow these steps to install <code>vs-granite</code> on your Mac/Linux machine :</p>
+    <ol>
+        <li>Clone the repository:</li>
+        <pre><code>git clone https://github.com/redhat-developer/vscode-granite.git </code></pre>
+        <li>Navigate to the project directory:</li>
+        <pre><code>cd vscode-granite</code></pre>
+        <li>Install the NPM dependencies:</li>
+        <pre><code>npm run install:ci</code></pre>
+    </ol>
+</body>
+
+This will install the dependencies for the extension and its [webview(s)](https://code.visualstudio.com/api/extension-guides/webview)
+
+#### Use Vscode-granite UI to install extention and models
+
+This project offers an intuitive and streamlined UI designed to simplify the installation and management of extensions and Granite models. This user-friendly interface enables developers to quickly set up and configure their environment without needing to dive into complex configurations or terminal commands
+
+ step 1: **Run the Extenstion**
+
+Open VSCode and navigate to the Debugging tab on the left sidebar. Click on "Run and Debug," then select "Run Extension" to launch the extension.
+
+![runextension](/media/installandrun.gif)
+ 
+ step 2:  **Install ollama**
+
+Once the extension is running, a new window will prompt you to install Ollama. Simply click the "Install with home brew"(for mac users) button to initiate the installation process.
+
+![installollama](media/installollama.gif)
+
+step 3: **Install granite models** 
+
+ select the Granite model you wish to install. Follow the on-screen instruction to complete the setup of your models.
+
+![installmodels](media/installmodels.gif)
```

Comment:
Once the models are pulled into Ollama, Continue will be configured automatically to use them, the Continue chat view will open.